open-company/open-company-api
# Section update authorship

https://trello.com/c/PyfjkhgZ

To test:

Make some normal API requests with no authorization, note you get a 401:

```
curl -i -X GET http://localhost:3000/companies
```

```
curl -i -X GET http://localhost:3000/companies/buffer
```

Now make the same API calls as a logged in user in the UI which passes JWT credentials and note that you can see the company list and the buffer data.

Now edit one of the text sections of the buffer data. Then refresh the page (there's a UI bug) and see that the update was attributed to your logged in user. Pull the same section from the Clojure API and see that the `:author` is you:

```clojure
(section/get-company "buffer" "update")
```

Next rerun the sample data import (with delete flag) for `buffer.edn`, `mature.edn`, `new.edn` and `yoyo.edn`. Notice they have authors that aren't Stuart? Super.

Study the code with a critical eye. All is well? Merge.

After deploying to staging, run the import with delete on staging for `buffer.edn`, `mature.edn`, `new.edn` and `yoyo.edn`.
